factory/wsSovereignNotifierFactory.go

```diff
@@ -9,20 +11,31 @@ import (
 	"github.com/multiversx/mx-chain-sovereign-notifier-go/process/notifier"
 )
 
+const addressLen = 32
+
 // CreatWsSovereignNotifier will create a ws sovereign shard notifier
 func CreatWsSovereignNotifier(cfg config.Config) (process.WSClient, error) {
-	sovereignNotifier, err := notifier.NewSovereignNotifier(cfg)
+	marshaller, err := factory.NewMarshalizer(cfg.WebSocketConfig.MarshallerType)
```

Comment:
can be added as param on Create method

Reply:
ok.. with them received as param, this component could have reached 100% coverage in tests

Reply:
This is not a component. This is "main" func, or the "creator of all components"

Reply:
Similar to our stateless `nodeRunner`

Reply:
ok understood